go/mcap/reader.go

```diff
@@ -66,6 +66,8 @@ type Reader struct {
 
 type MessageIterator interface {
 	Next([]byte) (*Schema, *Channel, *Message, error)
+	ReadNextInto(buf []byte, msg *Message) ([]byte, error)
+	SchemaAndChannelForID(channelID uint16) (*Schema, *Channel, error)
```

Comment:
explain why we need this - alternatively, consider an API where you populate the message and also return the schema and channel pointers in one call?